### DISCUSSION TITLE
Populating over fields behaves inconsistently between collections and refs

### DISCUSSION THREAD

**Discussion by co-sic** [mikro-orm/mikro-orm#6045] [General]
### Describe the bug

If i use the fields parameter in a query to populate a one-to-many relation, then the collection will be loaded with all the fields of the entity. This is also what typescript suggests, and what i would expect.
If i use the fields parameter to populate a one-to-one relation, then the populated entity will only have the id field. But typescript tells me that it has all the field, even with using ref and $. 

I'm not sure what is the correct behaviour for fields and if you can even use it to just "populate" a whole relation, the documentation is a little light on that part. 

### Reproduction

https://github.com/co-sic/node-mikro-orm-playground/blob/main/src/app.ts

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.10

### Node.js version

v20.14.0

### Operating system

ubuntu

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
The `fields` hint is capable of populating relations, but it's just a fallback, I would rather suggest staying with `populate` hint instead and using `fields` only for partial loading. Putting collection properties in there feels just weird to me, since those don't represent any column directly. For to-one relations like in your test, this should indeed select just the FK and not the whole entity, that's exactly the difference between `fields: ['owner']` and `populate: ['owner']` and is the correct behaviour. The fact that this breaks the return type is sad, but I am not entirely sure if this is worth fixing, the typing there is already way too complex. 

Btw I would also appreciate reproductions in the form of single-file tests (as in https://github.com/mikro-orm/reproduction).

**Reply by co-sic** [reply]
What about using fields and populate together in one query, is this supported behaviour? I think i found a bug then using both together, i can try to make a reproduction for this, if you say this is supported. If it's not supported, then this is a little limiting, since i can then either use populate or fields. Sometimes i want to populate some deeply nested entity, but not everything along the way. Was there a feature for fields to select everything with `fields: ['owner.*']` at some point? I thought i saw something like this in the docs at some point, but could not find it anymore.

Otherwise I understand that fields should not be used for populating collections and that the types are extremly complex already. For me i would be a big benefit of either using only fields for everything (also populating) or at least having the types be correct if you use it like in the examle above. In a bigger project with a lot of people working on it this can help a lot to catch errors on the type level already. But i understand that this is not a priority.

And i will use the single file tests for future reproductions, thanks :+1:

**Reply by B4nan** [reply]
Yes, you can use both options at the same time. If you specify only the `fields` hint, the `populate` hint is inferred from that - so internally there are both options in place effectively, even if you use just `fields`. 

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/EntityManager.ts#L2072

Using `owner.*` should work, but looks like we only have one test for select-in strategy, it indeed seems to be broken with the joined one.

**Reply by B4nan** [reply]
This should help with that: 1f0dbd4efdd47875368e31db10e4e6f981adc323

**Reply by co-sic** [reply]
Okay, nice to know i did not imagine seeing this `owner.*` usage somewhere. But i could not find it in the current docs. Thanks for the fix! I have 1 case of combination of populate field that gives a wierd result and one usage of fields on a collection that also behaves not normal (like `books.title`). Im on vaccation now for 2 weeks but i will get a simple reproduction test for each case and post them here after i'm back.

**Reply by B4nan** [reply]
So even with the fix above it's not working correctly? I'll wait for the repro, no rush.
